Add FT232R symbol/component and FT232RL device

MartijnBraam opened the pull request:
This adds the FT232R symbol and the FT232RL device (with the SSOP28 package that it requires)

![screenshot from 2018-12-03 00-01-40](https://user-images.githubusercontent.com/6928199/49345515-48122780-f686-11e8-913e-6e25ad55006d.png)

ubruhin commented:
Thanks @MartijnBraam :smiley: 

Here my comments after a quick review:

- The new component category is very deep nested, I'd suggest to either move the new category one level up (as child of "Interface"), or use the "Drivers, Receivers, Transmitters" (without subcategory) instead. Or what do you think @dbrgn?
- Symbol: Text elements for "{{NAME}}" and "{{VALUE}}" are missing. Name should be at the top left, and value at the bottom left of the symbol's outline.
- Symbol: Generally for ICs we try to avoid placing pins at top and bottom of a symbol. We think schematics are more clean with pins only at left and right of IC symbols. Especially for GND it's often useful to place it 5.08mm below VCC because that way you can directly connect capacitors to them :)
- Package: As it is a generic package, it would be good to move it into the base library. @dbrgn Could we generate that package with the parts generator?

ubruhin commented on cmp/fc58b79f-8fbf-45d3-b97c-2faaa02279f6/component.lp:
```diff
@@ -0,0 +1,117 @@
+ (deprecated false)
+ (category 58601391-795b-4ebf-ae55-984352167ce4)
+ (schematic_only false)
+ (default_value "{{PARTNUMBER}}")
```
For manufacturer-specific components, the following default value is recommended:

```suggestion
 (default_value "{{PARTNUMBER or DEVICE or COMPONENT}}")
```

ubruhin commented on cmp/fc58b79f-8fbf-45d3-b97c-2faaa02279f6/component.lp:
```diff
@@ -0,0 +1,117 @@
+ (category 58601391-795b-4ebf-ae55-984352167ce4)
+ (schematic_only false)
+ (default_value "{{PARTNUMBER}}")
+ (prefix "IC")
```
For ICs we use the prefix "U":

```suggestion
 (prefix "U")
```

MartijnBraam replied:
I'll update it

I used this because that's what used in the opamp example so I guess the docs need an update: https://docs.librepcb.org/#gettingstarted-libraries-sym

ubruhin replied:
Good point, I created an issue in the docs repo: https://github.com/LibrePCB/librepcb-doc/issues/27

dbrgn commented:
> @dbrgn Could we generate that package with the parts generator?

Yes, I can give it a try next week if I find time. We should be able to parametrize the SOIC generator for SSOP support. (I actually already started with generating TSSOP a while ago, but the branch is still sitting on my computer...)

MartijnBraam committed:
Added FT232R symbol and FT232RL device

MartijnBraam commented:
I've updated the symbol so there's space for the caps and crystal you might need.

![screenshot from 2018-12-03 01-24-37](https://user-images.githubusercontent.com/6928199/49346390-f7083080-f691-11e8-8295-16b760e2b802.png)

The previous symbol was made to closely match the symbol in the FTDI datasheet.

MartijnBraam committed:
Fixed reset line on FT232R component

MartijnBraam committed:
Updated component info

MartijnBraam committed:
Fixed pin positions on symbol

MartijnBraam committed:
Moved the component category a level up

MartijnBraam committed:
Moved pins some more

ubruhin commented:
Thanks for the update!

Actually the new category should be moved to the LibrePCB Base library. Could you maybe open a pull request there? The category's directory can just be moved, no need for any adjustments.

Regarding the symbol:
- Text elements should be left-aligned to the corners of the outline (corresponding library convention is work in progress, see draft [here](https://docs.librepcb.org/_branches/conventions/#libraryconventions-symbols-textelements)).
- I would move the TEST pin 2.54mm to bottom, just because there is enough space available :)

Component and device look good so far :+1:

MartijnBraam committed:
Moved test pin and fixed label alignment

MartijnBraam committed:
Removed component category from this package

MartijnBraam commented:
I've fixed the text elements and the test pin and moved the category.

Its probably a good idea to use a generated SSOP28 package instead of the one I've drawn since I'm not 100% sure everything aligns correctly and the silkscreen from the generated packages are nicer.

dbrgn commented:
Assigning this to me, so I remember to finish the SOP generator :slightly_smiling_face:

dbrgn commented:
TSSOP work ongoing at https://github.com/LibrePCB-Libraries/LibrePCB_Base.lplib/pull/33. Adding SSOP afterwards should be easy.

dbrgn commented:
SSOP packages done: https://github.com/LibrePCB-Libraries/LibrePCB_Base.lplib/pull/35

dbrgn commented:
Hi @MartijnBraam 

SSOP footprints have finally(!) been added to the base library. We have re-used your UUIDs, so if you simply delete your own package and make sure to have a dependency on the base library, everything should work as intended.

ubruhin committed:
Remove package SSOP28

It is now available in the Base library.

ubruhin committed:
Add "LibrePCB Base" library as dependency

ubruhin commented:
I removed the SSOP28 package and added the base library as dependency, so we can merge this now.

Thanks again @MartijnBraam, and sorry for taking such a long time to get this merged :grin:
